x/tokenfactory/types/denoms.go

```diff
@@ -66,7 +66,7 @@ func DeconstructDenom(denom string) (creator string, subdenom string, err error)
 	// So we have to join [2:] with a "/" as the delimiter to get back the correct subdenom which should be "atomderivative/sikka"
 	subdenom = strings.Join(strParts[2:], "/")
 
-	return creator, subdenom, nil
+	return addr.String(), subdenom, nil
```

Comment:
Just curious - what difference does this make for mutation testing?

Reply:
yeah this change is unclear to me. We can rename it as "creatorAddr" if we want to, but this generally seems odd to me

Reply:
Hrmm, this particular instance seems like a non-issue, but I would like to get mutation testing fixed to not make this error in the future.

This is good to merge rn though! Nice work